app/models/need_revision.rb

```diff
@@ -8,7 +8,7 @@ class NeedRevision
 
   default_scope -> { order_by(:created_at.desc) }
 
-  belongs_to :need
+  belongs_to :need, touch: false, inverse_of: :revisions
```

Comment:
TBH, I dunno.  I'll try it and see (I was trying all the things in one go).  It'll be better to introduce the smallest change that avoids the problem.